uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/rules/RemoveBrokersBeforeTestRule.java

```diff
@@ -41,22 +42,19 @@ public Statement apply(final Statement base, final Description description) {
         return new Statement() {
             @Override
             public void evaluate() throws Throwable {
-                final BrokerMicrosoftAuthenticator authenticator = new BrokerMicrosoftAuthenticator();
-                authenticator.uninstall();
-
-                // Auth App may still be installed if device admin (Samsung devices for instance)
-                if (authenticator.isInstalled()) {
-                    TestContext.getTestContext().getTestDevice().getSettings().disableAdmin(DeviceAdmin.MICROSOFT_AUTHENTICATOR);
-                    authenticator.uninstall();
-                }
+                final List<ITestBroker> testBrokers = CommonUtils.getAllPossibleTestBrokers();
 
-                final BrokerCompanyPortal companyPortal = new BrokerCompanyPortal();
-                companyPortal.uninstall();
+                for (final ITestBroker broker : testBrokers) {
+                    broker.uninstall();
 
-                // CP may still be installed if device admin
-                if (companyPortal.isInstalled()) {
-                    TestContext.getTestContext().getTestDevice().getSettings().disableAdmin(DeviceAdmin.COMPANY_PORTAL);
-                    companyPortal.uninstall();
+                    if (broker.isInstalled()) {
+                        // The broker app will still be installed on the device if it is enabled
+                        // as a device admin. In this case, we need to disable the admin and then
+                        // uninstall.
+                        TestContext.getTestContext().getTestDevice()
+                                .getSettings().disableAdmin(broker.getAdminName());
```

Comment:
nit: Hm.  Would it break if we just disabled it whether or not if needs it?

Reply:
Yeah, you actually can't disable it if it is NOT ad admin. This is disabling admin via UI....and the button for **disabling** would be _disabled_ if not an admin. It will break the tests.